hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergedReadHandle.java

```diff
@@ -99,7 +99,7 @@ private Option<FileSlice> getLatestFileSlice() {
         && hoodieTable.getMetaClient().getCommitsTimeline().filterCompletedInstants().lastInstant().isPresent()) {
       return Option.fromJavaOptional(hoodieTable
           .getHoodieView()
-          .getLatestMergedFileSlicesBeforeOrOn(partitionPathFileIDPair.getLeft(), instantTime)
+          .getLatestFileSlices(partitionPathFileIDPair.getLeft())
```

Comment:
can it return uncommitted file slices? I think it won't but just to confirm.

Reply:
It won't, the method intends to return the latest commit but it invokes the `getLatestMergedFileSlicesBeforeOrOn` which has some problems on NB-CC scenarios.